src/VisualStudio/Core/Def/Implementation/Utilities/BrowserHelper.cs

```diff
@@ -29,14 +34,29 @@ public static bool TryGetUri(string link, out Uri uri)
             return true;
         }
 
-        public static Uri CreateBingQueryUri(string errorCode, string title)
+        public static Uri CreateBingQueryUri(DiagnosticData diagnostic)
         {
-            if (string.IsNullOrWhiteSpace(title))
-            {
-                return new Uri(string.Format(BingSearchString, errorCode));
-            }
+            var errorCode = diagnostic.Id;
+            var title = diagnostic.ENUMessageForBingSearch;
+
+            string userId;
+            string language;
+            string projectType;
+            diagnostic.Workspace.GetLanguageAndProjectTypeAndUser(diagnostic.ProjectId, out userId, out language, out projectType);
+
+            return CreateBingQueryUri(errorCode, title, userId, language, projectType);
+        }
+
+        public static Uri CreateBingQueryUri(string errorCode, string title, string userId, string language, string projectType)
+        {
+            errorCode = errorCode ?? string.Empty;
+            title = title ?? string.Empty;
+            language = language ?? string.Empty;
+            projectType = projectType ?? string.Empty;
+            userId = userId ?? string.Empty;
 
-            return new Uri(string.Format(BingSearchString, errorCode + " " + title));
+            var url = string.Format(BingSearchString, Uri.EscapeDataString(title), Uri.EscapeDataString(language), Uri.EscapeDataString(projectType), Uri.EscapeDataString(RequestId), Uri.EscapeDataString(userId), Uri.EscapeDataString(errorCode));
```

Comment:
Can we simplify this and remove all the places where we are plumbing the userID through. We should make minimally necessary changes. We can always add the plumbing for userid later.